sdk/iot/Azure.Iot.Hub.Service/tests/ConfigurationsClientTests.cs

```diff
@@ -0,0 +1,150 @@
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Azure.Iot.Hub.Service.Tests
+{
+    /// <summary>
+    /// Test all APIs of the ConfigurationsClient.
+    /// </summary>
+    public class ConfigurationsClientTests : E2eTestBase
+    {
+        /// <summary>
+        /// All this test class does is to make sure the call comes back with a response.
+        /// This test is not responsible to make sure the values that come back are accurate as that would be testing the service logic.
+        /// </summary>
+        public ConfigurationsClientTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        /// <summary>
+        /// Test basic lifecycle of a Twin Configuration.
+        /// This test includes CRUD operations only.
+        /// </summary>
+        [Test]
+        public async Task ConfigurationsClient_ConfigurationsLifecycle()
+        {
+            string testConfigurationId = $"configlifecycle{GetRandom()}";
+            // Generate a random priority less than 100
+            int testPriority = int.Parse(GetRandom());
+            IotHubServiceClient client = GetClient();
+            TwinConfiguration twinConfiguration = CreateTestConfig(testConfigurationId);
+            TwinConfiguration createdConfig;
+
+            try
+            {
+                // Create a twin configuration
+                Response<TwinConfiguration> createResponse =
+                    await client.Configurations.CreateOrUpdateConfigurationAsync(twinConfiguration);
+                createdConfig = createResponse.Value;
+
+                // Get twin configuration
+                Response<TwinConfiguration> getResponse = await client.Configurations.GetConfigurationAsync(testConfigurationId).ConfigureAwait(false);
+
+                getResponse.Value.Etag.Should().BeEquivalentTo(createdConfig.Etag, "ETag value should not have changed.");
+                createdConfig = createResponse.Value;
+
+                // Update a configuration
+                createdConfig.Priority = testPriority;
+                Response<TwinConfiguration> updatedConfig = await client.Configurations.CreateOrUpdateConfigurationAsync(createdConfig, IfMatchPrecondition.UnconditionalIfMatch).ConfigureAwait(false);
+                Assert.AreEqual(updatedConfig.Value.Priority, testPriority, "Priority should have been updated.");
+            }
+            finally
+            {
+                // Delete twin configuration
+                await Cleanup(client, twinConfiguration);
+            }
+        }
+
+        /// <summary>
+        /// Test get configurations
+        /// </summary>
+        [Test]
+        public async Task ConfigurationsClient_GetConfigurations()
+        {
+            const int configurationsCount = 5;
+            TwinConfiguration[] twinConfigurations = new TwinConfiguration[configurationsCount];
+            TwinConfiguration[] createdConfigurations = new TwinConfiguration[configurationsCount];
```

Comment:
You can just use a list and do .Add below. Generally a more commonly used data type.